projects/igniteui-angular/src/lib/grids/columns/column.component.ts

```diff
@@ -2118,7 +2118,7 @@ export class IgxColumnComponent implements AfterContentInit, OnDestroy, ColumnTy
         }
 
         if (hasIndex) {
-            grid._moveColumns(this, targetColumn);
+            index === grid._pinnedColumns.length - 1 ? grid._moveColumns(this, targetColumn, 1) : grid._moveColumns(this, targetColumn, 0);
```

Comment:
You should use `DropPosition` enum when setting the drop position inside the `_moveColumns` method instead of plain numbers